### ISSUE TITLE
Intermediate layers do not make use of gamma parameters

### ISSUE THREAD

**Issue by nlml** [rinuboney/ladder#13]
Hi there,

First of all thanks for this implementation, I have been making good use of it in my efforts to study the ladder network.

One thing I noticed, in [line 130](https://github.com/rinuboney/ladder/blob/master/ladder.py#L130) of the current ladder.py, you have:
`h = tf.nn.relu(z + weights["beta"][l-1])`

shouldn't this be:
`h = tf.nn.relu(weights['gamma'][l-1] * (z + weights["beta"][l-1]))`
? (As per equation 10 [here](https://arxiv.org/pdf/1511.06430.pdf))

Regards,
Liam

**Comment by rinuboney** [comment]
Good to hear that you found it useful! As you can find mentioned in page 6 of the paper - [Semi-Supervised Learning with Ladder Networks](https://arxiv.org/abs/1507.02672), the beta and gamma parameters of batch normalization are redundant in case of some activations functions. In case of the ReLU activation function, the scaling parameter gamma is redundant.

**Comment by nlml** [comment]
Ah okay, make sense. Thanks!
